src/cargo/sources/registry/mod.rs

```diff
@@ -827,6 +847,22 @@ impl<'gctx> Source for RegistrySource<'gctx> {
                     shell.note("if possible, try a compatible non-yanked version")?;
                 }
             }
+            if precise_prerelease_in_use && self.gctx.cli_unstable().precise_pre_release {
+                let name = dep.package_name();
+                let version = req
+                    .precise_version()
+                    .expect("--precise <prerelease-version> in use");
+                if self
+                    .selected_precise_prerelease
+                    .insert((name, version.clone()))
+                {
+                    let mut shell = self.gctx.shell();
+                    shell.warn(format_args!(
+                        "selected prerelease package `{name}@{version}`"
+                    ))?;
+                    shell.note("if possible, try a compatible non-prerelease version")?;
```

Comment:
Not sure if we want a message for prereleases. A yanked version implies a "don't use me" message from the crate author. Prereleases are in a different story like "use me at your own risk but thanks for being a tester!" Also, a prerelease version is more visible than a yanked non-prerelease version. People using it may already notice.

Reply:
I agree. We can add it back if needed.